### PR TITLE
Fix build with sundials 7

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
Could you review this one @PeterMeisrimelModelon ?
It fixes several issues for building with sundials 7.x
Also fixes the cvode NULL pointer bug shown by the tests and the nfcnjacs number test

File: src/lib/sundials_callbacks.pxi [hunk 1]
```diff
@@ -63,7 +69,7 @@ cdef inline void arr2nv_inplace(x, N_Vector out) noexcept:
     cdef void* data_ptr=PyArray_DATA(ndx)
     memcpy((<N_VectorContent_Serial>out.content).data, data_ptr, n*sizeof(realtype))
     
-cdef inline np.ndarray nv2arr(N_Vector v) noexcept:
```
Comment (modelonrobinandersson) [inline src/lib/sundials_callbacks.pxi:66]: Isn't `noexcept` necessary for Cython 3.x?

Comment (jschueller) [reply src/lib/sundials_callbacks.pxi:66]: not for functions returning pure python objects

**Comment by jschueller** [comment]
would it be possible to have a 3.5.1 after that ? it would ease packaging

File: src/lib/sundials_callbacks.pxi (hunk 1 above)
Comment (modelonrobinandersson) [reply src/lib/sundials_callbacks.pxi:66]: Ahh, I see, thank you!

**Review by modelonrobinandersson — APPROVED** [review]
@jschueller I think we are just missing a changelog entry, other than that it looks good to me.

**Comment by jschueller** [comment]
@modelonrobinandersson done, and added one last commit to fix latex formulas escape (avoids SyntaxWarning)

**Comment by modelonrobinandersson** [comment]
> @modelonrobinandersson done, and added one last commit to fix latex formulas escape (avoids SyntaxWarning)

looks good to me!

**Comment by jschueller** [comment]
good for you too @PeterMeisrimelModelon ?

File: examples/kinsol_ors.py [hunk 2]
```diff
@@ -18,8 +18,9 @@
 import os
 import nose
 import numpy as np
-import scipy as sp
 import scipy.sparse as sps
+import scipy.sparse.linalg as spsl
```
Comment (PeterMeisrimelModelon) [inline examples/kinsol_ors.py:22]: Is there any particular motivation for this change?

Comment (jschueller) [reply examples/kinsol_ors.py:22]: yes, scipy.sparse.linalg is a standalone module thats not imported by scipy.sparse so the following does not work:
```
import scipy.sparse as sps
# cannot use sps.linalg
```
instead we must import scipy.sparse.linalg separately

**Pull request merged**